crypto/asn1/x_bignum.c

```diff
@@ -130,9 +130,16 @@ static int bn_c2i(ASN1_VALUE **pval, const unsigned char *cont, int len,
 static int bn_secure_c2i(ASN1_VALUE **pval, const unsigned char *cont, int len,
                          int utype, char *free_cont, const ASN1_ITEM *it)
 {
+    int ret;
+    BIGNUM *bn;
+
     if (!*pval)
         bn_secure_new(pval, it);
-    return bn_c2i(pval, cont, len, utype, free_cont, it);
+    ret = bn_c2i(pval, cont, len, utype, free_cont, it);
+    /* Set constant-time flag for all secure BIGNUMS */
+    bn = (BIGNUM *)*pval;
```

Comment:
You need to check the return value of bn_c2i,
see the bn_c2i calls bn_free in error handling.

Reply:
Ok, working on it.